File: problems/122/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        if not prices:
            return 0
        
        profit = 0
        
        for i in range(len(prices) - 1):
            if prices[i] < prices[i  + 1]:
                profit += prices[i + 1] - prices[i]
        
        return profit

File: problems/122/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int ans = 0;
        
        for (int i = 1; i < size(prices); i++) {
            // if (prices[i] > prices[i - 1]) {
            //     ans += prices[i] - prices[i - 1];
            // }
            ans += max(0, prices[i] - prices[i - 1]);
        }
        
        return ans;
    }
};